src/devices/bus/nscsi/cd.cpp

```diff
@@ -226,10 +226,10 @@ void nscsi_cdrom_device::scsi_command()
 		 * SUPPORTED.
 		 */
 		int page = scsi_cmdbuf[2];
-		int size = scsi_cmdbuf[4];
+		int size = (scsi_cmdbuf[3] << 8) | scsi_cmdbuf[4];
 		switch(page) {
 		case 0:
-			std::fill_n(scsi_cmdbuf, 36, 0);
+			std::fill_n(scsi_cmdbuf, size, 0);
```

Comment:
Is there a chance this could cause a buffer overrun?  You’re using a 16-but value (up to 65,535) with a 4,096-byte buffer.

Reply:
That's fair, I can't say there wouldn't be some weird bugged request that wouldn't possibly overflow. I didn't see any responses that would be widely huge (more than 255) in my research online so I think it'd be safe to revert the size back to just `scsi_cmdbuf[4]` unless a case is found that breaks.